File: examples/v1-archive/myapp/models/animal.go
package models

import (
	"fmt"
)

//line models/animal.zn:3
type AnimalImpl struct {
	Name  string
	Sound string
}

func (a *AnimalImpl) SetName(v string)  { a.Name = v }
func (a *AnimalImpl) GetSound() string  { return a.Sound }
func (a *AnimalImpl) SetSound(v string) { a.Sound = v }

type Animal interface {
	SetName(string)
	GetSound() string
	SetSound(string)
	Speak() string
	GetName() string
}

var _ Animal = (*AnimalImpl)(nil)

func NewAnimal(name string, sound string) *AnimalImpl {
	obj := &AnimalImpl{}
//line models/animal.zn:8
	obj.Name = name
//line models/animal.zn:9
	obj.Sound = sound
	return obj
}

func (a *AnimalImpl) Speak() string {
//line models/animal.zn:13
	return fmt.Sprintf("%v says %v!", a.Name, a.Sound)
}

func (a *AnimalImpl) GetName() string {
//line models/animal.zn:17
	return a.Name
}

File: zinc-out/Animal.java
import java.util.*;
import java.util.stream.*;

public class Animal {
    private String name = "";
    private String sound = "";
    
    
    public String getName() { return this.name; }
    public void setName(String name) { this.name = name; }
    public String getSound() { return this.sound; }
    public void setSound(String sound) { this.sound = sound; }
    public String speak() {
        return name + " says " + sound;
    }
    
}
